src/main/java/net/dv8tion/jda/core/requests/Route.java

```diff
@@ -393,12 +369,36 @@ public String toString()
         private final Route baseRoute;
         private final String ratelimitRoute;
         private final String compiledRoute;
+        private final boolean hasQueryParams; 
 
-        private CompiledRoute(Route baseRoute, String ratelimitRoute, String compiledRoute)
+        private CompiledRoute(Route baseRoute, String ratelimitRoute, String compiledRoute, boolean hasQueryParams)
         {
             this.baseRoute = baseRoute;
             this.ratelimitRoute = ratelimitRoute;
             this.compiledRoute = compiledRoute;
+            this.hasQueryParams = hasQueryParams;
+        }
+
+        private CompiledRoute(Route baseRoute, String ratelimitRoute, String compiledRoute)
+        {
+            this(baseRoute, ratelimitRoute, compiledRoute, false);
+        }
+
+        public CompiledRoute withQueryParams(String name, String value)
+        {
+            return new CompiledRoute(baseRoute, ratelimitRoute, compiledRoute + (hasQueryParams ? '&' : '?') + name + '=' + value, true);
+        }
+
+        public CompiledRoute withQueryParams(String... params)
+        {
```

Comment:
Adding a check for the first thing.

I've added the `(String, String)` version because it theoretically is faster (no loop). Micro-optimization :pray:. 
If you want I can also just simply remove it